api/provisioning/v1alpha1/provisioningrequest_types.go

```diff
@@ -65,9 +65,9 @@ type NodePoolRef struct {
 	// Contains the namespace of the created NodePool.
 	Namespace string `json:"namespace,omitempty"`
 	// Represents the timestamp of the first status check for hardware provisioning
-	HardwareProvisioningCheckStart metav1.Time `json:"hardwareProvisioningCheckStart,omitempty"`
+	HardwareProvisioningCheckStart *metav1.Time `json:"hardwareProvisioningCheckStart,omitempty"`
 	// Represents the timestamp of the first status check for hardware configuring
-	HardwareConfiguringCheckStart metav1.Time `json:"hardwareConfiguringCheckStart,omitempty"`
+	HardwareConfiguringCheckStart *metav1.Time `json:"hardwareConfiguringCheckStart,omitempty"`
```

Comment:
Apparently in Go [1.24](https://tip.golang.org/doc/go1.24) we are finally getting `omitzero` which potentially checks for the zero value or `IsZero()` and treat it nil automatically...huge pain point 

/lgtm

Reply:
@pixelsoccupied Exactly!